p2p/src/protocol/cbfmgr.rs

```diff
@@ -414,15 +414,21 @@ impl<F: Filters, U: SyncFilters + Events + Wakeup + Disconnect> FilterManager<F,
         }
     }
 
-    /// Rollback filter header chain by a given number of headers.
-    pub fn rollback(&mut self, n: usize) -> Result<(), filter::Error> {
-        self.filters.rollback(n)?;
-        self.schedule_tick();
-
-        let height = self.filters.height();
+    /// Rollback filters to the given height.
+    pub fn rollback(&mut self, height: Height) -> Result<(), filter::Error> {
+        // It's possible that a rollback doesn't affect the filter chain, if the filter headers
+        // haven't caught up to the block headers when the re-org happens.
+        if height >= self.filters.height() {
+            return Ok(());
+        }
 
         // Purge stale block filters.
         self.cache.rollback(height);
+        // Rollback filter header chain.
+        self.filters.rollback(height)?;
+
+        // TODO: Cancel requests for filter headers and fitlers? Or just ignore them when they
+        // come in?
```

Comment:
I like the idea to cancel the request, but ignoring it sounds good too if the number of requests is not too high.

Reply:
Yeah, so the problem with cancelling is knowing exactly which requests to cancel, since we request filter ranges. So might be too complicated.

Reply:
>Yeah, so the problem with cancelling is knowing exactly which requests to cancel, since we request filter ranges. So might be too complicated

Yep, I agree